### PR TITLE
Native YAML support

### LINKED ISSUE #132: Quit converting YAML to JSON for native YAML support in CloudFormation
CloudFormation has recently supported YAML templates natively:
https://aws.amazon.com/blogs/aws/aws-cloudformation-update-yaml-cross-stack-references-simplified-substitution/

`stack_master` however still converts YAML to JSON before uploading templates, which ignores tag-based syntax such as `!Ref` and causes template validation errors. 

So it would be great if `stack_master` quits converting YAML to JSON.

### PR DESCRIPTION
**Pull request by skatsuta**
Resolve #132 

My implementation is a bit primitive so any comments are welcome!
(For example, maybe it'd be better if I create a new `Template` class and make it have template body and format.)

### CODE REVIEW

**Comment by stevehodgkiss** [comment]
Looks good! 👍

File: lib/stack_master/stack.rb [hunk 1]
```diff
@@ -85,10 +92,17 @@ def self.generate(stack_definition, config)
           tags: stack_definition.tags,
           parameters: parameters,
           template_body: template_body,
+          template_format: template_format,
           notification_arns: stack_definition.notification_arns,
           stack_policy_body: stack_policy_body)
     end
 
+    def self.identify_template_format(template_body)
+      return :json if template_body =~ /^{/x # ignore leading whitespaces
```
Comment (jacobbednarz) [inline lib/stack_master/stack.rb]: do we have access to the file extension here? it might be more straight forward using that if we can.

**Comment by skatsuta** [comment]
Can I get merged? Or need additional reviews?
Thanks!

**Comment by stevehodgkiss** [comment]
I've released 0.8.0 with these changes. Thanks @skatsuta!

**Pull request merged**